string description
string type
string topicName
